### ISSUE TITLE
Would it be possible to use a Submenu Hub as a widget?

### ISSUE THREAD

**Issue by tjamieg** [jurialmunkey/skin.arctic.horizon#189]
As the title suggests, I have created some Submenu's and really like the look and functionality of the Submenu Hub (I think it's way cooler than the Showcase Hub)... but, would it be possible to use it as a widget instead of a Submenu? Beacuse, at the moment, I have a pretty blank screen and only get the Submenu Hub when I physically click into one of my main menu items as shown below.

![Startup](https://user-images.githubusercontent.com/48328397/83681186-941e8f00-a5d1-11ea-9289-d56d1c95a702.jpg)
![Home](https://user-images.githubusercontent.com/48328397/83681187-941e8f00-a5d1-11ea-9fbf-b2cdf7a41e07.jpg)
![TV](https://user-images.githubusercontent.com/48328397/83681188-94b72580-a5d1-11ea-8329-b68365ce6f16.jpg)
![Film](https://user-images.githubusercontent.com/48328397/83681190-94b72580-a5d1-11ea-8a4b-becf7715bea2.jpg)

**Comment by jurialmunkey** [comment]
The big roadblock is the auto width buttons. If I got rid of those, I could do it relatively easy - but I think that's part of the charm of the submenu hub in the first place...

**Comment by tjamieg** [comment]
I agree - I definitely prefer it with auto width buttons, than without.

But I think I could live without the auto width for a while if it meant being able to use the submenu hub as a widget for now, at least until you manage to find a fix which would allow them all to play nicely together anyway.

Also, I currently have my submenu / button-actions set to “noop” (as I have everything I need in the hub widget underneath each button); would you happen to know if there may be a different action I can use, so that when the button is pressed, the focus moves from the button down onto the widget?

**Comment by jurialmunkey** [comment]
> Also, I currently have my submenu / button-actions set to “noop” (as I have everything I need in the hub widget underneath each button); would you happen to know if there may be a different action I can use, so that when the button is pressed, the focus moves from the button down onto the widget?

You could use `Action(Down)` or just `Down` 😉 
https://kodi.wiki/view/Action_IDs

Alternatively, `SetFocus(5610)` should work if Down doesn't.


Also, you know that if you just set one widget in the "Widgets" of a home menu, you will get a layout similar to the hub (just without buttons) -- you could set your most used widget to that one. What you set in the "Widgets" section won't have any impact on the hub itself.

![Honeyview_screenshot003](https://user-images.githubusercontent.com/4618719/83712400-0aba9b80-a669-11ea-8d83-ca2e51a0eb84.jpg)

**Comment by tjamieg** [comment]
> You could use `Action(Down)` or just `Down` 😉
> https://kodi.wiki/view/Action_IDs
> 
> Alternatively, `SetFocus(5610)` should work if Down doesn't.

Thank you, I feel quite silly now, I didn't realise the answer would've been so simple!

> Also, you know that if you just set one widget in the "Widgets" of a home menu, you will get a layout similar to the hub (just without buttons) -- you could set your most used widget to that one. What you set in the "Widgets" section won't have any impact on the hub itself. That will get you most of the way to what you want.

I am aware of the single widget view, but I prefer everything to be in one place, as opposed to using both widgets AND submenu's (this could be quite confusing for some - "do I click or do I go right?"). I'm also aware that I could just use multiple widgets, which is what I did initially, but then you can't see which widgets you have off the page without scrolling down through each and every one.

The submenu hub, on the other hand, shows every widget category on one page, where you can just scroll straight across to the one you want. In my opinion this would be perfect if it could be displayed as a widget without having to click into the menu item.

**Comment by jurialmunkey** [comment]
Ah that makes sense - I figured that was reason but just though I should mention the single widget in case you didn't know about it.

**Comment by Fubuz** [comment]
To be able to have the submenu hub as a home widget would definately be really useful - like in the 3rd picture, but as a Widget - Home. Hope this feature makes into the skin. Fingers crossed :)

**Comment by tjamieg** [comment]
@jurialmunkey, any progress on this feature? Would it be a possibility nowadays? Thanks.
